lib/mongo_client.js

```diff
@@ -55,45 +132,62 @@ const validOptions = require('./operations/mongo_client_ops').validOptions;
  * @class
  * @param {string} url The connection URI string
  * @param {object} [options] Optional settings
+ * @param {string} [options.appname=undefined] The name of the application that created this MongoClient instance. MongoDB 3.4 and newer will print this value in the server log upon establishing each connection. It is also recorded in the slow query log and profile collections
```

Comment:
No need to document as `=undefined` as this is the default of an optional value. Just say `@param {string} [options.appname]`